lib/Dialect/SV/SVDialect.cpp

```diff
@@ -70,14 +70,17 @@ static llvm::ManagedStatic<StringSet<>, ReservedWordsCreator> reservedWords;
 /// keyword or any other name in the set \p recordNames. Use the int \p
 /// nextGeneratedNameID as a counter for suffix. Update the \p recordNames with
 /// the generated name and return the StringRef.
-StringRef circt::sv::resolveKeywordConflict(
-    StringRef origName, llvm::StringMap<size_t> &nextGeneratedNameIDs) {
+StringRef
+circt::sv::resolveKeywordConflict(StringRef origName,
+                                  llvm::StringMap<size_t> &nextGeneratedNameIDs,
+                                  bool caseInsensitiveKeywords) {
   // Get the list of reserved words we need to avoid.  We could prepopulate this
   // into the used words cache, but it is large and immutable, so we just query
   // it when needed.
 
   // Fast path: name is valid
-  if (!reservedWords->count(origName)) {
+  if (!reservedWords->count(caseInsensitiveKeywords ? origName.lower()
```

Comment:
Are reserved keywords always guaranteed to be all lower-case? I think so, but just checking :+1:.

Reply:
Looks like in `ReservedWords.def` we do have entries that aren't all lowercase, but those are macros.

Reply:
Yeah, this looks okay. I don't think we need to worry about collisions with macro names. I'm actually not even sure why we're checking for collisions against macro names in the first place? These are always guarded with a leading `` ` ``, e.g., `` `foo ``, and that _cannot_ cause a conflict.

I have seen lint tools complain about case insensitive keyword collisions, e.g., `reg foo` and `wire FOO` being in the same module. We will eventually have to add this option and maybe that is the time to fix/remove the macro collision checking.